src/main/scala/pipedsl/common/Utilities.scala

```diff
@@ -679,4 +697,66 @@ object Utilities {
   val is_handle_var :Id => Boolean =
     { id: Id => id.v.startsWith(lock_handle_prefix) }
 
+  val generic_type_prefix = "__SUPERSPECIALGENERIC_"
+  def is_generic(t :Any) :Boolean = t match {
+    case TNamedType(name) => name.v.startsWith(generic_type_prefix)
+    case TSizedInt(l, _) => is_generic(l)
+    case TBitWidthVar(name) => name.v.startsWith(generic_type_prefix)
+    case _:Type => false
+    case id:Id => id.v.startsWith(generic_type_prefix)
+    case s:String => s.startsWith(generic_type_prefix)
+  }
+
+  val not_gen_pref = "__NOT_GEN"
+  def degenerify(t :Type) :Type =
```

Comment:
Ok curious to see where this is used...

Reply:
well, it WAS used, and it will potentially be useful when i implement the more complicated type inference, so i'd like to keep it for now